engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/DebuggingEnsoTest.java

```diff
@@ -246,6 +269,86 @@ public void testHostValues() {
     }
   }
 
+  /**
+   * Both {@code Date.new 2024 12 15} and {@code Date.parse "2024-12-15"} should be seen by the
+   * debugger as the exact same objects. Internally, the value from {@code Date.parse} is a host
+   * value.
+   */
+  @Test
+  public void hostValueIsTreatedAsItsEnsoCounterpart() {
+    Value fooFunc =
+        createEnsoMethod(
+            """
+        from Standard.Base import Date, Date_Time, Dictionary
+        polyglot java import java.lang.String
+        polyglot java import java.util.List as JList
+        polyglot java import java.util.Map as JMap
+
+        foo _ =
+            d_enso = Date.new 2024 12 15
+            d_java = Date.parse "2024-12-15"
+            dt_enso = Date_Time.now
+            dt_java = Date_Time.parse "2020-05-06 04:30:20" "yyyy-MM-dd HH:mm:ss"
+            str_enso = "Hello_World"
+            str_java = String.new "Hello_World"
+            list_enso = [1, 2, 3]
+            list_java = JList.of 1 2 3
+            dict_enso = Dictionary.from_vector [["A", 1], ["B", 2]]
+            dict_java = JMap.of "A" 1 "B" 2
+            end = 42
+        """,
+            "foo");
+
+    try (DebuggerSession session =
+        debugger.startSession(
+            (SuspendedEvent event) -> {
+              switch (event.getSourceSection().getCharacters().toString().strip()) {
+                case "end = 42" -> {
+                  DebugScope scope = event.getTopStackFrame().getScope();
+
+                  DebugValue ensoDate = scope.getDeclaredValue("d_enso");
+                  DebugValue javaDate = scope.getDeclaredValue("d_java");
+                  assertSameProperties(ensoDate.getProperties(), javaDate.getProperties());
+
+                  DebugValue ensoDateTime = scope.getDeclaredValue("dt_enso");
+                  DebugValue javaDateTime = scope.getDeclaredValue("dt_java");
+                  assertSameProperties(ensoDateTime.getProperties(), javaDateTime.getProperties());
+
+                  DebugValue ensoString = scope.getDeclaredValue("str_enso");
+                  DebugValue javaString = scope.getDeclaredValue("str_java");
+                  assertSameProperties(ensoString.getProperties(), javaString.getProperties());
+
+                  DebugValue ensoList = scope.getDeclaredValue("list_enso");
+                  DebugValue javaList = scope.getDeclaredValue("list_java");
+                  assertSameProperties(ensoList.getProperties(), javaList.getProperties());
+
+                  DebugValue ensoDict = scope.getDeclaredValue("dict_enso");
+                  DebugValue javaDict = scope.getDeclaredValue("dict_java");
+                  assertSameProperties(ensoDict.getProperties(), javaDict.getProperties());
```

Comment:
But shouldn't the test also check for JavaScript, Python & co. objects? Maybe use `ValuesGenerator.dates()`, etc.?

Reply:
Tests for `py` and `js` objects added in 2c46980beecdddbd33a5ecaec87e416198edc83e